JUnit provider: KotlinNullPointerException when some response headers don't match

Version: pact-jvm-provider-junit_2.12 3.6.0 and master

**Steps to reproduce:**
1. Create a provider test
2. Test an interaction with at least 2 expected response headers
3. Make it pass
4. Introduce a mismatch in one of the response headers

**Expected outcome:**
A helpful error message

**Actual outcome:**
```
Verifying a pact between foo and bar
  Given with 3 tracks
  OPTIONS /v1/me/tracks
      Request Failed - null

Failures:

0) OPTIONS /v1/me/tracks
      kotlin.KotlinNullPointerException

com.snoutify.snoutify.PactIntegrationTest > snoutify - OPTIONS /v1/me/tracks FAILED

    java.lang.AssertionError: 

0 - null

java.lang.AssertionError: 
0 - null

	at au.com.dius.pact.provider.junit.target.BaseTarget.getAssertionError(BaseTarget.kt:71)
	at au.com.dius.pact.provider.junit.target.HttpTarget.testInteraction(HttpTarget.kt:80)
	at au.com.dius.pact.provider.junit.InteractionRunner$interactionBlock$statement$1.evaluate(InteractionRunner.kt:207)
	at au.com.dius.pact.provider.junit.RunStateChanges.evaluate(RunStateChanges.kt:17)
	at au.com.dius.pact.provider.junit.InteractionRunner.run(InteractionRunner.kt:143)
	at au.com.dius.pact.provider.junit.PactRunner.runChild(PactRunner.kt:137)
	at au.com.dius.pact.provider.junit.PactRunner.runChild(PactRunner.kt:51)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.runTestClass(JUnitTestClassExecutor.java:106)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:58)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:38)
	at org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass(AbstractJUnitTestClassProcessor.java:66)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:32)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:93)
	at com.sun.proxy.$Proxy2.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.processTestClass(TestWorker.java:117)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:155)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:137)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:404)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
	at java.lang.Thread.run(Thread.java:745)

    0 - null
```

Here's the pact:
```json
{
  "consumer": {
    "name": "foo"
  },
  "provider": {
    "name": "bar"
  },
  "interactions": [
    {
      "description": "OPTIONS /v1/me/tracks",
      "providerState": "with 3 tracks",
      "request": {
        "method": "OPTIONS",
        "path": "/v1/me/tracks",
        "headers": {
          "Origin": "http://localhost",
          "Access-Control-Request-Method": "GET",
          "Access-Control-Request-Headers": "authorization"
        }
      },
      "response": {
        "status": 200,
        "headers": {
          "Access-Control-Allow-Origin": "http://localhost",
          "Access-Control-Allow-Credentials": "true",
          "Access-Control-Allow-Headers": "authorization"
        }
      }
    }
  ],
  "metadata": {
    "pactSpecification": {
      "version": "2.0.0"
    }
  }
}
```

--- a/pact-jvm-provider-gradle/src/test/groovy/au/com/dius/pact/provider/gradle/ResponseComparisonTest.groovy
+++ b/pact-jvm-provider-gradle/src/test/groovy/au/com/dius/pact/provider/gradle/ResponseComparisonTest.groovy
@@ -19,7 +19,7 @@ class ResponseComparisonTest {
 
   @Before
   void setup() {
-    response = new Response(200, ['Content-Type': 'application/json'], OptionalBody.body('{"stuff": "is good"}'))
+    response = new Response(200, ['A': 'mismatch', 'Content-Type': 'application/json'], OptionalBody.body('{"stuff": "is good"}'))
     actualStatus = 200
     actualBody = '{"stuff": "is good"}'
     actualResponse = [contentType: ContentType.APPLICATION_JSON]

--- a/pact-jvm-provider/src/main/kotlin/au/com/dius/pact/provider/ResponseComparison.kt
+++ b/pact-jvm-provider/src/main/kotlin/au/com/dius/pact/provider/ResponseComparison.kt
@@ -50,7 +50,7 @@ class ResponseComparison(
           headerResult = expected.headers.orEmpty().mapValues { null }.toMutableMap()
       } else {
         expected.headers.orEmpty().forEach { headerKey, _ ->
-          if (headerMismatchers[headerKey]!!.isNotEmpty()) {
+          if (headerMismatchers.containsKey(headerKey) && headerMismatchers[headerKey]!!.isNotEmpty()) {
               headerResult[headerKey] = headerMismatchers[headerKey]!!.first().mismatch
           } else {
               headerResult[headerKey] = null

